var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -61,14 +61,24 @@ def mpi_dependent_options(self):
                 errors = ['incompatible variants given'] + errors
                 raise RuntimeError('\n'.join(errors))
         else:
-            compiler_opts = [
-                '--with-mpi=1',
-                '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
-            ]
+            if self.compiler.name == "clang":
+                compiler_opts = [
+                    '--with-mpi=1',
+                    '--with-cc=%s  -Qunused-arguments' % join_path(self.spec['mpi'].prefix.bin, 'mpicc'), # Avoid confusing PETSc config by clang: warning: argument unused during compilation
+                    '--with-cxx=%s -Qunused-arguments' % join_path(self.spec['mpi'].prefix.bin, 'mpic++'),
+                    '--with-fc=%s' % join_path(self.spec['mpi'].prefix.bin, 'mpif90'),
+                    '--with-f77=%s' % join_path(self.spec['mpi'].prefix.bin, 'mpif77'),
+                ]
+            else:
+                compiler_opts = [
+                    '--with-mpi=1',
```

Comment:
This also enabled MPI unconditionally right?  It should depend on whether the spec has `+mpi`.